client/src/components/History/SwitchToHistoryLink.vue

```diff
@@ -37,21 +47,33 @@ const actionText = computed(() => {
     return "View in new tab";
 });
 
-function onClick(history: HistorySummary) {
+async function onClick(history: HistorySummary) {
     if (canSwitch.value) {
         if (props.filters) {
             historyStore.applyFilters(history.id, props.filters);
         } else {
-            historyStore.setCurrentHistory(history.id);
+            try {
+                await historyStore.setCurrentHistory(history.id);
+            } catch (error) {
+                Toast.error(errorMessageAsString(error));
+            }
         }
         return;
     }
     viewHistoryInNewTab(history);
 }
 
 function viewHistoryInNewTab(history: HistorySummary) {
-    const routeData = router.resolve(`/histories/view?id=${history.id}`);
-    window.open(routeData.href, "_blank");
+    if (
+        userOwnsHistory(userStore.currentUser, history) ||
+        history.published ||
```

Comment:
A history can also be directly shared with a user, that option is missing here.

Reply:
That's true, somehow i was able to access the History summary for the e.g. invocation I've shared, where I guess the History is inaccessible. I'll reduce this just to handling the setCurrentHistory error then

Reply:
The history for that invocation is public, you can see it here: https://usegalaxy.org/published/history?id=5abbe23547149481. It's true that you can't make this your current history (i.e. the history where new outputs would go to), but you can view it in a new tab.

Reply:
I think I somehow messed up with the conditions that check whether you can view it in another tab, and incorrectly Toasted the error then. Sorry about that, ill fix this.